sklearn/metrics/_regression.py

```diff
@@ -887,15 +905,15 @@ def _assemble_r2_explained_variance(
             avg_weights = None
         elif multioutput == "variance_weighted":
             avg_weights = denominator
-            if not np.any(nonzero_denominator):
+            if not xp.any(nonzero_denominator):
                 # All weights are zero, np.average would raise a ZeroDiv error.
                 # This only happens when all y are constant (or 1-element long)
                 # Since weights are all equal, fall back to uniform weights.
                 avg_weights = None
     else:
         avg_weights = multioutput
 
-    return np.average(output_scores, weights=avg_weights)
+    return xp.reshape(_average(output_scores, weights=avg_weights), (-1,))[0]
```

Comment:
I feel like this is quite cryptic compared to the existing line, like, why reshape and then taking the first element, when before we were just calling `np.average`